src/Middleware/Spa/SpaProxy/src/SpaProxyStartupFilter.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.SpaProxy
+{
+    internal class SpaProxyStartupFilter : IStartupFilter
+    {
+        public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
+        {
+            return builder =>
+            {
+                var env = builder.ApplicationServices.GetRequiredService<IWebHostEnvironment>();
+                if (env.IsDevelopment())
```

Comment:
We need to stop using IsDevelopment in framework libraries, it should only be used in application code since they're the ones that define the environments. This injection logic should be controlled by user settable options instead.
https://github.com/dotnet/aspnetcore/issues/28937

Reply:
This check is here as an additional safeguard to ensure this never gets used outside of development. There is already another check that happens to prevent this injection in the first place, this was done as an additional check.

In general the issue here is that we don't have a great way to determine when you are in a `development environment` (aka running with `dotnet run`) over some other environment.

Things like this package, or the runtime support for Static Web Assets are specifically tailored to enable development scenarios and we want to go as far as possible to avoid involving them outside of those contexts.

Reply:
I've removed the check since its an additional guarantee, that said, I've left some comments on the issue, we keep "complaining" about this without offering a solution.

Reply:
> There is already another check that happens to prevent this injection in the first place, this was done as an additional check.

What's the other check?

The proposal for 6.0 is to change the framework usage of IsDevelopment so there is user targeted option for each feature and we fall back to IsDevelopment if the option is not set. How you specify that option does depend on the feature setup.

Reply:
What's the workaround for someone that does rename their dev environment?
`builder.UseMiddleware<SpaProxyMiddleware>();` is internal so they can't add it manually.

Reply:
All the services only get setup if there's a file in the app running folder, which gets generated at build time and not copied to the publish folder.

https://github.com/dotnet/aspnetcore/pull/32997/files#diff-1c1e51e26ffaf3976061683dbfd13701a7184b1d8d1191169aeda59b3003a97aR22

Reply:
The way this is plugged in is via a HostingStartup assembly in launchSettings.json and a file we generate during build. You don't opt-in to individual pieces. It is just meant to launch the CLI automatically and redirect the user to the proxy url (the ServerUrl, which is always localhost:XXX).

There's no public API on purpose since this is just a convenience to keep parity with the old SPA template experience.